components/cluster.py

```diff
@@ -7,6 +7,7 @@
 import dash_html_components as html
 import dash_core_components as dcc
 from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
```

Comment:
This isn't used, as a few of the other imports, too. Maybe go through them and remove the unused ones